fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1387,21 +1393,25 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
         }
 
         StringBuilder whereClauseSegment = new StringBuilder();
+        if (!endOfChain) {
+            whereClauseSegment.append(WHERE);
+        }
 
-        String valuesTable = QuerySegmentAggregator.tableName(resourceType.getSimpleName(), queryParm);
+        // If not a concrete resource type, then this is a system-wide search, so use the param table alias instead of the table name,
+        // which will be replaced with the table name for each resource type in QuerySegmentAggregator.buildWhereClause(...)
+        String valuesTable = ModelSupport.isAbstract(resourceType) ? paramTableAlias : QuerySegmentAggregator.tableName(resourceType.getSimpleName(), queryParm);
 
         // Build this piece of the segment. Use EXISTS instead of SELECT DISTINCT for much better performance
-        // missing:     NOT EXISTS (SELECT 1 FROM Observation_STR_VALUES V WHERE V.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID)
-        // not missing:     EXISTS (SELECT 1 FROM Observation_STR_VALUES V WHERE V.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID)
-        whereClauseSegment.append(WHERE);
+        // missing:     NOT EXISTS (SELECT 1 FROM Observation_STR_VALUES WHERE Observation_STR_VALUES.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID)
+        // not missing:     EXISTS (SELECT 1 FROM Observation_STR_VALUES WHERE Observation_STR_VALUES.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID)
         if (missing == null || missing) {
             whereClauseSegment.append(NOT);
         }
         whereClauseSegment.append(EXISTS);
 
         whereClauseSegment.append("(SELECT 1 FROM " + valuesTable + WHERE);
-        this.populateNameIdSubSegment(whereClauseSegment, queryParm.getCode(), valuesTable.toString());
-        whereClauseSegment.append(AND).append(valuesTable).append(".LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID"); // correlate the [NOT] EXISTS subquery
+        this.populateNameIdSubSegment(whereClauseSegment, queryParm.getCode(), valuesTable);
+        whereClauseSegment.append(AND).append(valuesTable).append(".LOGICAL_RESOURCE_ID = ").append(logicalRsrcTableAlias).append(".LOGICAL_RESOURCE_ID"); // correlate the [NOT] EXISTS subquery
```

Comment:
Just curious if there was a reason you switched from RESOURCES table to LOGICAL_RESOURCES table here

Reply:
Because with a whole-system search, only the LR table is visible in the encompassing sub-query, so using R doesn't work. LR is visible in either.